File: src/main/java/com/gitlab/lae/intellij/actions/tree/ui/ActionList.java
package com.gitlab.lae.intellij.actions.tree.ui;

import com.intellij.ui.components.JBList;

import java.awt.event.KeyEvent;
import java.util.Collection;

public class ActionList extends JBList<ActionPresentation> {

    public ActionList(Collection<ActionPresentation> items) {
        super(items);
        setCellRenderer(new ActionPresentationRenderer());
    }

    @Override
    protected void processKeyEvent(KeyEvent e) {
        e.setSource(this);
        super.processKeyEvent(e);
    }

}

File: src/main/kotlin/com/gitlab/lae/intellij/actions/tree/ui/ActionList.kt
package com.gitlab.lae.intellij.actions.tree.ui

import com.intellij.ui.components.JBList
import java.awt.event.KeyEvent

open class ActionList(items: Collection<ActionPresentation>) : JBList<ActionPresentation>(items) {

  init {
    cellRenderer = ActionPresentationRenderer()
  }

  public override fun processKeyEvent(e: KeyEvent) {
    e.source = this
    super.processKeyEvent(e)
  }
}
